### DISCUSSION TITLE
Easy way to populate all relationships under a given relationship?

### DISCUSSION THREAD

**Discussion by squivix** [mikro-orm/mikro-orm#3712] [Ideas]
Is there a way to populate all relationships under a given relationship?

I know we can populate all relationships by passing `{populate: true}`, but can we do the same with all relationships under a specific relationship?

Instead of

```ts
const tags = await em.findAll(BookTag, { populate: ['books.title', 'books.author' 'books.publisher'] });
```

I want to do:

```ts
const tags = await em.findAll(BookTag, { populate: {books:true} });
```

and that should populate everything under books.

**Comment by B4nan** [comment]
It might be possible to do this, although it won't type check so you need the type cast:

```ts
const tags = await em.find(BookTag, {}, { populate: [{ field: 'books', children: [{ all: true }] }] } as any);
```

Or with a separate `em.populate` call, you can do this:

```ts
const tags = await em.find(BookTag, {}, { populate: ['books'] });
await em.populate(tags.flatMap(t => t.books.getItems()), { populate: true });
```

Would be good to have more native way to do it, but I don't want to allow passing in objects again, it complicates the inference a lot.
